src/ir/auth_logic/souffle_emitter.h

```diff
@@ -102,14 +122,48 @@ class SouffleEmitter {
     return absl::StrCat(".decl ", predicate.name(), "(", arguments, ")");
   }
 
-  std::string EmitDeclarations() {
-    std::vector elements_vec(std::make_move_iterator(declarations_.begin()),
-                             std::make_move_iterator(declarations_.end()));
-    std::sort(elements_vec.begin(), elements_vec.end());
-    return absl::StrJoin(elements_vec, "\n",
-                         [this](std::string* out, auto decl) {
-                           return absl::StrAppend(out, EmitDeclaration(decl));
-                         });
+  std::string EmitArguments(const std::vector<datalog::Argument>& arguments) {
+    return absl::StrJoin(
+        arguments, ", ", [](std::string* out, datalog::Argument argument) {
+          return absl::StrAppend(out, argument.argument_name(), " : ",
+                                 argument.argument_type().name());
+        });
+  }
+
+  std::string EmitRelationDeclarations(
+      const datalog::Program& program,
+      const absl::flat_hash_set<std::string>& skip_declarations) {
+    std::vector<std::string> declaration_strings;
+    for (const auto& declaration : program.relation_declarations()) {
+      if (absl::c_find(skip_declarations, declaration.relation_name()) !=
+          skip_declarations.end())
```

Comment:
Why `c_find` rather than just calling `skip_declarations.find(declaration.relation_name())`?

Reply:
Just curious. Is `skip_declarations.find(..)` better in terms of performance than `c_find`?

Reply:
Yes, I think it just `c_find` does a linear search. `skip_declarations.find` will be amortized constant time as `skip_declarations` is hash set.

Reply:
Yes, I think `c_find` does a linear search. `skip_declarations.find` is like a hash table lookup, which should be amortized constant time.

Reply:
I see. Thanks! Made code changes to use `skip_declaration.find(..)`

Reply:
I see. Thanks! Made code changes to use `skip_declaration.find(..)`